lib/galaxy/config.py

```diff
@@ -363,9 +365,9 @@ def __init__( self, **kwargs ):
             # so dependending on the context a better default can be used (
             # request url in a web thread, Docker parent in IE stuff, etc...)
             galaxy_infrastructure_url = "http://localhost"
-            port = self.guess_galaxy_port()
-            if port:
-                galaxy_infrastructure_url += ":%s" % (port)
+            web_port = self.galaxy_infrastructure_web_port or self.guess_galaxy_port()
+            if web_port:
+                galaxy_infrastructure_url += ":%s" % (web_port)
```

Comment:
Great thanks. Hopefully this complicated configuration nonsense is just needed for these dynamic applications ([Vagrant VM for Galaxy-IPython now available](https://github.com/jmchilton/galaxy-ipython-machine#using-the-machine)) and static installations will be able to just specify a fixed ``galaxy_infrastructure_url`` that will work over clusters of Docker hosts.